File: Array/128. Longest Consecutive Sequence.java
class Solution {
    HashSet<Integer> numberList = new HashSet<Integer>();
    public int longestConsecutive(int[] nums) {
        if (nums.length==0) return 0;
        for(int x:nums){
            numberList.add(x);
        }
        int max = 1;
        for(int i = 0;i<nums.length;i++){
            int temp = nums[i];
            int localmax = 1;
            while(numberList.contains(temp+1)){
                temp++;
                localmax++;
                numberList.remove(temp);
            }
            temp = nums[i];
            while(numberList.contains(temp-1)){
                temp--;
                localmax++;
                numberList.remove(temp);
            }
            max = (localmax>max) ? localmax : max;
        }
        return max;
    }
}

File: Array/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashSet = set()
        for num in nums:
            hashSet.add(num)
        
        maxCount=0
        for num in nums:
            now = num
            Count = 0
            while now in hashSet:
                hashSet.remove(now)
                Count=Count+1
                now=now-1
            now = num+1
            while now in hashSet:
                hashSet.remove(now)
                Count=Count+1
                now=now+1
            maxCount=max(Count,maxCount)
        return maxCount